neqo-transport/src/cc/classic_cc.rs

```diff
@@ -562,153 +519,186 @@ mod tests {
         )
     }
 
-    fn persistent_congestion(lost_packets: &[SentPacket]) -> bool {
-        let mut cc = ClassicCongestionControl::new(NewReno::default());
-        for p in lost_packets {
-            cc.on_packet_sent(p);
+    fn congestion_control(cc: CongestionControlAlgorithm) -> Box<dyn CongestionControl> {
+        match cc {
+            CongestionControlAlgorithm::NewReno => {
+                Box::new(ClassicCongestionControl::new(NewReno::default()))
+            }
+            CongestionControlAlgorithm::Cubic => {
+                Box::new(ClassicCongestionControl::new(Cubic::default()))
+            }
         }
+    }
 
-        cc.on_packets_lost(Some(now()), None, PTO, lost_packets);
-        if cc.cwnd() == CWND_INITIAL / 2 {
-            false
-        } else if cc.cwnd() == CWND_MIN {
-            true
-        } else {
-            panic!("unexpected cwnd");
+    fn persistent_congestion(lost_packets: &[SentPacket], persistent_expected: bool) {
+        for cc_alg in &[
+            CongestionControlAlgorithm::NewReno,
+            CongestionControlAlgorithm::Cubic,
+        ] {
+            let mut cc = congestion_control(*cc_alg);
+            for p in lost_packets {
+                cc.on_packet_sent(p);
+            }
+
+            cc.on_packets_lost(Some(now()), None, PTO, lost_packets);
+
+            let expected_cwnd_no_persistent_congestion = match cc_alg {
+                CongestionControlAlgorithm::NewReno => CWND_INITIAL / 2,
+                CongestionControlAlgorithm::Cubic => {
+                    CWND_INITIAL * CUBIC_BETA_USIZE_QUOTIENT / CUBIC_BETA_USIZE_DIVISOR
+                }
+            };
+            let persistent = if cc.cwnd() == expected_cwnd_no_persistent_congestion {
+                false
+            } else if cc.cwnd() == CWND_MIN {
+                true
+            } else {
+                panic!("unexpected cwnd");
+            };
+            assert_eq!(persistent, persistent_expected);
         }
     }
```

Comment:
```suggestion
    fn persistent_congestion_by_algorithm(cc_alg: CongestionControlAlgorithm, reduced_cwnd: usize, lost_packets: &[SentPacket], persistent_expected: bool) {
        let mut cc = congestion_control(*cc_alg);
        for p in lost_packets {
            cc.on_packet_sent(p);
        }

        cc.on_packets_lost(Some(now()), None, PTO, lost_packets);
        let persistent = if cc.cwnd() == reduced_cwnd {
             false
        } else if cc.cwnd() == CWND_MIN {
            true
        } else {
            panic!("unexpected cwnd");
        };
        assert_eq!(persistent, persistent_expected);
    }

    fn persistent_congestion(lost_packets: &[SentPacket], persistent_expected: bool) {
        persistent_congestion_by_algorithm(CongestionControlAlgorithm::NewReno, CWND_INITIAL / 2, lost_packets, persistent_expected);
        persistent_congestion_by_algorithm(CongestionControlAlgorithm::Cubic, CWND_INITIAL * CUBIC_BETA_USIZE_QUOTIENT / CUBIC_BETA_USIZE_DIVISOR, lost_packets, persistent_expected);
    }
```

Reply:
We were talking past each other. I will take this suggestion.